tests/Aspire.Hosting.Oracle.Tests/OracleFunctionalTests.cs

```diff
@@ -0,0 +1,408 @@
+
+            var oracle = builder.AddOracle("oracle");
+            var db = oracle.AddDatabase(oracleDbName);
+
+            var ready = builder;
+
+            if (init)
+            {
+                oracle.WithInitBindMount(bindMountPath);
+            }
+            else
+            {
+                oracle.WithDbSetupBindMount(bindMountPath);
+            }
+
+            using var app = builder.Build();
+
+            await app.StartAsync();
+
+            await app.WaitForText("Completed: ALTER DATABASE OPEN", cancellationToken: cts.Token);
+
+            var hb = Host.CreateApplicationBuilder();
+
+            hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+            {
+                [$"ConnectionStrings:{db.Resource.Name}"] = await db.Resource.ConnectionStringExpression.GetValueAsync(default)
+            });
+
+            hb.AddOracleDatabaseDbContext<TestDbContext>(db.Resource.Name);
+
+            using var host = hb.Build();
+
+            try
+            {
+                await host.StartAsync();
+
+                // Wait until the database is available
+                await pipeline.ExecuteAsync(async token =>
+                {
+                    var dbContext = host.Services.GetRequiredService<TestDbContext>();
+                    var databaseCreator = (RelationalDatabaseCreator)dbContext.Database.GetService<IDatabaseCreator>();
+                    return await databaseCreator.CanConnectAsync(token);
+                }, cts.Token);
+
+                var dbContext = host.Services.GetRequiredService<TestDbContext>();
+
+                var brands = await dbContext.Cars.ToListAsync(cancellationToken: cts.Token);
+                Assert.Single(brands);
+                Assert.Equal("BatMobile", brands[0].Brand);
+            }
+            finally
+            {
+                await app.StopAsync();
+            }
+        }
+        finally
+        {
+            try
+            {
+                Directory.Delete(bindMountPath, true);
+            }
+            catch
+            {
+                // Don't fail test if we can't clean the temporary folder
+            }
+        }
+    }
+
+    private TestDistributedApplicationBuilder CreateDistributedApplicationBuilder()
+    {
+        // Do not use the test container registry as Oracle has their own
+        var builder = TestDistributedApplicationBuilder.Create();
+        builder.Services.AddLogging(config => config.SetMinimumLevel(LogLevel.Information));
+        builder.Services.AddXunitLogging(testOutputHelper);
+        builder.Services.AddHostedService<ResourceLoggerForwarderService>();
+        return builder;
+    }
+
+    private static IResourceBuilder<OracleDatabaseServerResource> ConfigureTestOracleDatabase(IResourceBuilder<OracleDatabaseServerResource> oracle)
+    {
+        return oracle
+            .WithImage("gvenzl/oracle-free", "23-slim-faststart")
```

Comment:
This doesn't seem like an ideal test situation. The intention of these tests is to ensure that our end-to-end works correctly. If we are using a different docker image in our tests than what customers will use when they just `builder.AddOracle("oracle");` then we aren't really testing how our customers will use the product. If there is a change/update in the oracle docker container Aspire.Hosting.Oracle uses, this test won't catch it.